### PR TITLE
REACH-154 Nodes with Array current values should show their contents, not just simply "Array"

### PR DESCRIPTION
**Pull request by marimano**
When node's value section is hovered it shows specified number of array items and if not all items are shown "..." will be shown. In the node's config section we can choose how many array items we want to see. For this field (number of shown items) a strong validation is implemented (it should be integer number from 1 to array length)

It requires [changes in Dynamo](https://github.com/Benglin/Dynamo/pull/256)

### CODE REVIEW

File: app/scripts/views/NodeViews/Base.js [hunk 1]
```diff
@@ -1,5 +1,61 @@
...
+
+      e && e.stopPropagation();
+  }
+
+  function keyDown (e) {
+      console.log(e.keyCode);
+      var zeroCode = 48, nineCode = 57,
+          rightSidedZeroCode = 96, rightSidedNineCode = 105;
+      // backspace, shift, ctrl, 4 arrows, delete
+      var allowedKeyCodes = [8, 16, 17, 37, 38, 39, 40, 46];
```
Comment (marimano) [inline app/scripts/views/NodeViews/Base.js]: allow this buttons as well to give a user a possibility of moving cursor, selecting and deleting typed number

File: app/app.html [hunk 2]
```diff
@@ -401,8 +412,15 @@
...
 
-          </ul>
-        </div>
+            <li class="dropdown-header shown-items hidden">Shown array items</li>
+
+            <li class="node-dropown-menu-item shown-items hidden">
+                <input type="number" class="shown-items" min="1" max="10" value="10" step="1">
+                <span class="shown-items" style="padding-left: 5px">of 1 is shown</span>
+            </li>
+
```
Comment (marimano) [inline app/app.html]: this part of the config section will be hidden if the node doesn't have array items (not array or empty array)

**Comment by marimano** [comment]
Hi @pboyer, I've updated this PR with receiving array items in a separate response. By default node request first 10 items from Dynamo. Then if a user chooses more than 10 items to show Flood request only items that are currently absent to show and after receiving appends them into the list.
If Flood hasn't received yet requested items and the user chooses more elements than there are currently present but less than ones were requested new request won't be sent.
Please take a look

**Comment by pboyer** [comment]
Hi @marimano, 

I tried this branch for a while.  The contents of the nodes is just "List ...".  Not sure why.  Please get this up to date and I'll take another look.

**Comment by marimano** [comment]
Hi @pboyer I pulled the latest changes from your branch, resolved all conflicts and corrected some inaccuracies. Node content wasn't being updated because of lack of sending ArrayItemsMessage.
So please take a look again.

Hope you'll be not so angry because of changes not related to arrays :) I just corrected code I had got with the latest pull :)

File: app/scripts/views/NodeViews/Num.js [hunk 3]
```diff
@@ -69,21 +69,23 @@ define(['backbone', 'underscore', 'jquery', 'BaseNodeView', 'jqueryuislider'], f
 
       // adjust settings dropdown so that it stays open while editing
       // doesn't select the node when you're editing
-      $('.dropdown.keep-open').on({
+      this.$el.find('.dropdown.keep-open').on({
         "shown.bs.dropdown": function() {
           this.selectable = false;
           this.model.set('selected', false);
-          $('.dropdown.keep-open').data('closable', false);
+          this.$el.find('.dropdown.keep-open').data('closable', false);
```
Comment (marimano) [inline app/scripts/views/NodeViews/Num.js]: `$('.dropdown.keep-open')` will give elements from all nodes when we need only one from current node

**Pull request merged**